File: java/darts/src/main/java/Darts.java
import static java.lang.Math.pow;

class Darts {

    double distance;
    Darts(double x, double y) {
        distance = pow(pow(x, 2) + pow(y, 2), 0.5);
    }

    int score() {
        return distance > 10 ? 0 : distance > 5 ? 1 : distance > 1 ? 5 : 10;
    }

}

File: kotlin/darts/src/main/kotlin/Darts.kt
import kotlin.math.ceil
import kotlin.math.pow
import kotlin.math.sqrt

object Darts {

    fun score(x: Number, y: Number) =
        when (ceil(sqrt(x.toDouble().pow(2) + y.toDouble().pow(2))).toInt()) {
            in 0..1 -> 10
            in 2..5 -> 5
            in 6..10 -> 1
            else -> 0
        }
}
